ruby/ql/src/queries/security/cwe-295/RequestWithoutValidation.ql

```diff
@@ -15,6 +15,20 @@ import ruby
 import codeql.ruby.Concepts
 import codeql.ruby.DataFlow
 
-from HTTP::Client::Request request, DataFlow::Node disablingNode
-where request.disablesCertificateValidation(disablingNode)
-select request, "This request may run with $@.", disablingNode, "certificate validation disabled"
+from
+  HTTP::Client::Request request, DataFlow::Node disablingNode, DataFlow::Node origin, string ending
+where
+  request.disablesCertificateValidation(disablingNode, origin) and
+  // Showing the origin is only useful when it's a different node than the one disabling
+  // certificate validation, for example in `requests.get(..., verify=arg)`, `arg` would
+  // be the `disablingNode`, and the `origin` would be the place were `arg` got its
+  // value from.
+  //
+  // NOTE: We compare the locations instead of DataFlow::Nodes directly, since for
+  // snippet `Excon.defaults[:ssl_verify_peer] = false`, `disablingNode = argumentNode`
+  // does NOT hold.
+  if disablingNode.getLocation() = origin.getLocation()
+  then ending = "."
+  else ending = " by the value from $@."
+select request, "This request may run without certificate validation because it is $@" + ending,
```

Comment:
I think this will result in an invalid results pattern error if we ever hit the `else` case, since there will be two `$@` components in the message string.

Reply:
It actually just works :sparkles: Using the Python version as example, and the [Python test-file](https://github.com/github/codeql/blob/9cb249fc2f7bc1043d50f7d9424026a3a68e4f2a/python/ql/test/query-tests/Security/CWE-295-RequestWithoutValidation/make_request.py).

# VS Code

![image](https://user-images.githubusercontent.com/1054041/186170045-03a819dd-b46c-466b-92a6-e4020e14367e.png)

# Code Scanning

![image](https://user-images.githubusercontent.com/1054041/186170141-b3ae6d68-5117-4c9f-987a-c30142a3b070.png)


![image](https://user-images.githubusercontent.com/1054041/186170164-81450f49-8149-456d-8f12-daf47f9b93d8.png)

Reply:
Oh cool, sorry for doubting you!

Reply:
> Oh cool, sorry for doubting you!

No worries. It's good to question whether things work before just blindly accepting suspicious stuff like this. I also remember questioning whether I could even do this (I had not seen a query do this before), but it just works :muscle: